ngraph/core/src/op/color_convert_nv12.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/color_convert_nv12.hpp"
+
+#include <memory>
+#include <ngraph/validation_util.hpp>
+
+#include "itt.hpp"
+#include "ngraph/runtime/reference/color_convert_nv12.hpp"
+#include "openvino/core/layout.hpp"
+
+static const size_t N_DIM = 0;
+static const size_t H_DIM = 1;
+static const size_t W_DIM = 2;
+static const size_t C_DIM = 3;
+
+OPENVINO_RTTI_DEFINITION(ov::op::v8::ConvertColorNV12, "ConvertColorNV12", 0);
+
+ov::op::v8::ConvertColorNV12::ConvertColorNV12(const Output<Node>& arg, ColorConversion format)
+    : Op({arg}),
+      m_format(format),
+      m_single_tensor(true) {
+    constructor_validate_and_infer_types();
+}
+
+ov::op::v8::ConvertColorNV12::ConvertColorNV12(const Output<Node>& arg_y,
+                                               const Output<Node>& arg_uv,
+                                               ColorConversion format)
+    : Op({arg_y, arg_uv}),
+      m_format(format),
+      m_single_tensor(false) {
+    constructor_validate_and_infer_types();
+}
+
+void ov::op::v8::ConvertColorNV12::validate_and_infer_types() {
+    NGRAPH_OP_SCOPE(v8_Convert_NV12_validate_and_infer_types);
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0) == element::u8 || get_input_element_type(0) == element::f32,
+                          "Y input shall have u8 or f32 precision, got ",
+                          get_input_element_type(0));
+    const auto& shape_y = get_input_partial_shape(0);
+    NODE_VALIDATION_CHECK(this,
+                          shape_y.rank().is_static() && shape_y.rank().get_length() == 4,
+                          "Y input shall not have fully dynamic shape and have 'NCHW' layout");
+
+    NODE_VALIDATION_CHECK(this,
+                          shape_y[C_DIM].is_dynamic() || shape_y[C_DIM].get_length() == 1,
+                          "Y channels dimension shall be either dynamic or equal to 1. Current value is ",
```

Comment:
why do we need to allow dynamic channels? I believe it's always 1 while other dimensions can be dynamic

Reply:
Fixed in merged PR #7601